### PR TITLE
Make templatetags return None instead of 'unknown'

### PR DESCRIPTION
**Pull request by danielquinn**
This addresses #67 by dropping the default of returning  and instead returning .  As a result this should probably be seen as a breaking change, so I'm not sure how thrilled you might be about that.  Additionally, I added support for  to return either the browser or the os if not both, and added tests for the lot.

Ooh, and I also added `__pycache__` to .gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@
 /.tox/
 
 /docs/_build/
+__pycache__
```
Comment (Bouke) [inline .gitignore]: Can you revert this change? Please only include on-topic changes. You can add `__pycache__` to your global gitignore, this'll reduce duplication of such ignores.

Comment (danielquinn) [inline .gitignore]: Sure no problem, though it's a little weird that a Python project wouldn't ignore __pycache__ directories by default.  GitHub provides a pretty good default .gitignore file [that I used for my Paperless project](https://github.com/danielquinn/paperless/blob/master/.gitignore) if you wanna have a look.  (Obviously, the latter part of the file is stuff I added that's Paperless-specific).

**Comment by danielquinn** [comment]
I've updated the PR to include changes to the template, so let me know if there's something else you'd like to roll in.

**Comment by Bouke** [comment]
I've merged your PR, thank you for contributing.
